### PR TITLE
Add draft phylogenetics intro

### LINKED ISSUE #163: New tutorial: tskit for phylogeneticists / phylogenetics
As part of the website design, we want material to help people coming from the phylogenetic world, who might not realise that `tskit` can be used as a fast, scalable phylogenetic toolkit. It's not just tutorial material: there will presumably be some blurb etc on the website, but a tutorial is a good resource to have for this. There's a slight start in the "numba" stuff: https://github.com/tskit-dev/tutorials/issues/63.

@szhan might be able to contribute thoughts below, since he's coming from the phylogenetics world.

### PR DESCRIPTION
**Pull request by hyanwong**
Fixes #163 and a bit of #191

### CODE REVIEW

**Review by jeromekelleher — COMMENTED** [review]
This is really great, thanks @hyanwong!

One thing I would add is that we store mutations (and optionally the reference sequence) along with the tree, giving a single unified representation of the dataset comprising the tree(s), sequence data and metadata.

**Comment by hyanwong** [comment]
> One thing I would add is that we store mutations (and optionally the reference sequence) along with the tree, giving a single unified representation of the dataset comprising the tree(s), sequence data and metadata.

Great point. I'll add this and some other minor improvements. 

I don't know if we want to merge this yet, or wait until we have nice newick parsing to demo? Perhaps we can get some feedback on the content as it is, and then merge with the to-dos in there (which, to be fair, are present in some of the other released tutorials too).

**Comment by hyanwong** [comment]
> > One thing I would add is that we store mutations (and optionally the reference sequence) along with the tree, giving a single unified representation of the dataset comprising the tree(s), sequence data and metadata.
> 
> Great point. I'll add this and some other minor improvements.

Thinking about it, I reckon this is less of a "sticking point" for phylogeneticists and more of a "really nice to have" thing, so I'll not add it to the list of "Hints", but mention it at the top, and create  new section at the bottom.

**Comment by jeromekelleher** [comment]
SGTM. I think we can merge it whenever. We have plenty TODOs elsewhere, and we'll probably forget about it if we don't merge soon.

**Comment by benjeffery** [comment]
PIng me when I should review here. Better to merge with todos than let it fester.

**Comment by hyanwong** [comment]
@benjeffery - this is ready for review, with a few "todo"s in place. Some are dependent on fixing issues I've been opening in tskit projects.

File: phylogen.md
Comment (benjeffery) [inline phylogen.md:97]: typo? missing word after "entire"?

**Review by benjeffery — APPROVED** [review]
LGTM, one nit.
We need to put out a tsconvert release soon? Based on your needing to install from git?

File: phylogen.md
Comment (hyanwong) [reply phylogen.md:97]: Well spotted, thanks! I have a few more additions too. I'll add that to them and push as a separate commit so it's easy to see.

**Comment by jeromekelleher** [comment]
I don't really want to get into releasing tsconvert, installing from git is fine for now and if we see uptake we can shine things up a bit. Hopefully we'll have some real trees available soon in tskit form that we can use as examples.

**Comment by hyanwong** [comment]
OK, I think this is ready to merge. With the latest additional commit, most of it is filled out now, apart from one section that I will ask @szhan about, and fill in later.

It's in a good enough state to link to from the main tskit site, I think.

**Comment by jeromekelleher** [comment]
LGTM, merge whenever you'r happy

**Comment by hyanwong** [comment]
> LGTM, merge whenever you'r happy

Thanks. I don't have permissions to add auto merge labels to this repo. I guess it would be sensible to give me those permissions, as I'm doing lots of the tutorial stuff.

**Pull request merged**
